### DISCUSSION TITLE
BirdNetPi with Caddy behind a reverse proxy?

### DISCUSSION THREAD

**Discussion by bcardi0427** [mcguirepr89/BirdNET-Pi#825] [Q&A]
I have an Nginx Reverse Proxy setup on my network to deliver http and https requests to servers on my network. I know I could use the Caddy to do this in BirdNetPi but I don't want the Pi taking that traffic so I have an Nginx setup as a reverse proxy for my whole network. My network has 1 dynamic IP address but I have that arranged to switch the IP when needed with my Domain Host. I can't seem to get Caddy to accept anything from the Reverse Proxy and get errors unless using my network IP address. Is there anything in Caddy I have to do to have it accept the url from a reverse proxy? I can't seem to find anything but using caddy as the reverse proxy.

**Comment by gerdesj** [comment]
In the advanced settings there is a box to enter the URL that is used to access BirdNET-Pi.  Set that to the dynamic DNS name.  If that does not work then please post your nginx server.conf

Caddy is a very capable proxy itself and can do auto SSL (eg Lets Encrypt) out of the box just by putting a name in its config.  However as you are invested with nginx, let's get that sorted 8)

**Reply by bcardi0427** [reply]
I found the trusted_proxy statement for Caddy BUT it is in 2.6.3 and higher versions of Caddy and right now on Birdnetpi I have 2.6.2 of caddy. If I update Caddy to 2.6.3 or 2.6.4 will that break what we already have? The big problem with Caddy is by default it doesn't trust any proxies so you need to use the trusted_proxy statement to let it know what proxy to trust. They now have a global var so the trusted_proxy only needs to be done once instead of for each server. My new question is will updating Caddy break BirdNetPi?

**Comment by lloydbayley** [comment]
Interesting discussion. I have two birdnets running locally. One in the shed up in the back yard and one in the front yard to get the birds in the valley. I've been wondering if I could share them both to the internet with caddy without having to use nginx just for two installs when I read that caddy will do it for you.
What I'm wondering is, will it handle two https:// ok on different subdomains or does this make things "unpleasant' ?

**Comment by lloydbayley** [comment]
ooh, ooh, I think I might have managed it. Will report in later

Update: Yep, I managed it. Was overthinking things. Sorry, didn't mean to hijack this thread at all but thought it was somewhat related. As you were! :)
(Happy to share if anyone wants to do it through caddy)
